src/Physics/ini_model_DR.f90

```diff
@@ -659,6 +688,53 @@ SUBROUTINE friction_LSW6(DISC,EQN,MESH,BND)
 
   END SUBROUTINE friction_LSW6    ! Initialization of friction for bimaterial linear slip weakening
 
+!> Initialization of TP grid, Fourier coefficients and initial temperature/pressure for TP
+  !<
+  SUBROUTINE thermalPress_init(DISC,EQN)
+  !-------------------------------------------------------------------------!
+  IMPLICIT NONE
+  !-------------------------------------------------------------------------!
+  TYPE(tDiscretization), target  :: DISC
+  TYPE(tEquations)               :: EQN
+  !-------------------------------------------------------------------------!
+  ! Local variable declaration
+  INTEGER                        :: j
+  REAL                           :: dlDwn, hwid, Dwnmax, Dwnmax_norm
+  REAL, PARAMETER                :: pi=3.141592653589793
+  !-------------------------------------------------------------------------!
+  INTENT(INOUT) :: DISC,EQN
+  !-------------------------------------------------------------------------!
 
+  !values currently from bicycle code -> how can we optimize that?
```

Comment:
what is a bicycle code?

Reply:
I guess Stephanie referred to this as the spectral BIEM by Noda & Lapusta 2010

Reply:
ok I guess bi-cycle (two cycles?)